File: src/main/java/io/vertx/starter/Launcher.java
package io.vertx.starter;

import io.vertx.core.Vertx;
import io.vertx.core.logging.Logger;
import io.vertx.core.logging.LoggerFactory;

/**
 * @author Richard Gomez
 */
public class Launcher extends io.vertx.core.Launcher {

  private static final Logger LOG = LoggerFactory.getLogger(Launcher.class);

  public static void main(String[] args) {
    new Launcher().dispatch(args);
  }

  @Override
  public void afterStartingVertx(Vertx vertx) {
    super.afterStartingVertx(vertx);

    LOG.info("Hello, Vert.x Launcher!");
  }
}

File: src/main/kotlin/io/vertx/starter/Launcher.kt
package io.vertx.starter

import io.vertx.core.Launcher
import io.vertx.core.Vertx
import io.vertx.core.logging.Logger
import io.vertx.core.logging.LoggerFactory

/**
 * @author Richard Gomez
 */
class Launcher : Launcher() {

  companion object {
    val LOG: Logger = LoggerFactory.getLogger(Launcher::class.java)
  }

  override fun afterStartingVertx(vertx: Vertx?) {
    super.afterStartingVertx(vertx)

    LOG.info("Hello, Vert.x Launcher!")
  }
}
